streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceiverSupervisorImpl.scala

```diff
@@ -162,11 +169,11 @@ private[streaming] class ReceiverSupervisorImpl(
   }
 
   override protected def onStart() {
-    blockGenerator.start()
+    registeredBlockGenerators.foreach { _.start() }
```

Comment:
Yes, I am preserving the existing behavior where the default block generator is active as long as the supervisor is active, where as for the receiver, the custom block generators are started and stopped every time the receiver is started and stopped.